added list-meetings endpoint

Parimala032 opened the pull request:
## Description
This pull request implements the list-meetings functionality into the Webex Lyte application. 

## Jira Ticket
[SPARK-526689](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-526689)

Parimala032 committed:
Add files via upload

Added files index.html, app.js, style.css

Parimala032 committed:
Delete index.html

Parimala032 committed:
Add files via upload

mkesavan13 commented on app.js:
```diff
@@ -0,0 +1,131 @@
+    function displayMeetingDetails(meeting) {
+        meetingDetailsElement.innerHTML = ''; // Clear previous details
+
+        const detailsHTML = `
+            <p><strong>Meeting Link:</strong> ${meeting.webLink}</p>
+            <p><strong>Meeting Number:</strong> ${meeting.meetingNumber}</p>
+            <p><strong>Host Key:</strong> ${meeting.hostKey}</p>
+            <p><strong>Password:</strong> ${meeting.password}</p>
+            <p><strong>Access Code:</strong> ${meeting.telephony.accessCode}</p>
+            <p><strong>Sip Address:</strong> ${meeting.sipAddress}</p>
+        `;
```
Let's do this one line after another like this,

**Meeting Link**
https://cisco.webex.com/kmadavan

**Meeting Number**
543210045

**Host Key**
234567

.
.
.

**SIP Address**
kmadavan.cisco@webex.com

mkesavan13 commented on app.js:
```diff
@@ -0,0 +1,131 @@
+// Prerequisite data SIP and Access Token(Hard Coded)
+const myAccessToken = "ODcwYWE1MDQtODc5ZC00ZDhiLWI5ODYtZTk0ZGUyNTA4YzM2MTIwNmIzOTQtZDhm_PF84_1eb65fdf-9643-417f-9974-ad72cae0e10f";
+const sip = "pasoma.cisco@webex.com";
```
No need for a default sip address here

mkesavan13 commented on app.js:
```diff
@@ -0,0 +1,131 @@
+// Prerequisite data SIP and Access Token(Hard Coded)
+const myAccessToken = "ODcwYWE1MDQtODc5ZC00ZDhiLWI5ODYtZTk0ZGUyNTA4YzM2MTIwNmIzOTQtZDhm_PF84_1eb65fdf-9643-417f-9974-ad72cae0e10f";
```
This is a public repository. Let's avoid pushing any access tokens here-on. Not safe to do it. Please leave it empty and whoever uses shall fill theirs

mkesavan13 commented on app.js:
```diff
@@ -0,0 +1,131 @@
+        const joinButton = document.createElement('button');
+        joinButton.innerText = 'Join Meeting';
+        joinButton.classList.add('btn', 'join-btn');
+        joinButton.addEventListener('click', () => {
```
Let's leave this action empty for @RAM1232 to work on.

mkesavan13 commented on app.js:
```diff
@@ -0,0 +1,131 @@
+
+        ongoingMeetingListElement.innerHTML = '';
+
+        ongoingMeetings.forEach(meeting => {
+            const meetingElement = document.createElement('div');
+            meetingElement.classList.add('meeting');
+
+            meetingElement.onclick = () => displayMeetingDetails(meeting);
+
+            const meetingTitle = document.createElement('h2');
+            meetingTitle.textContent = meeting.title || 'No title provided';
+            meetingElement.appendChild(meetingTitle);
+
+            const meetingDate = document.createElement('p');
+            meetingDate.textContent = `Date: ${new Date(meeting.start).toLocaleString()}`;
+            meetingElement.appendChild(meetingDate);
+
+            const meetingOrganizer = document.createElement('p');
+            meetingOrganizer.textContent = `Organizer: ${meeting.hostDisplayName || 'Unknown'}`;
+            meetingElement.appendChild(meetingOrganizer);
+
+            ongoingMeetingListElement.appendChild(meetingElement);
+        });
```
We could keep creation of the list as a separate function so that it can be used both for ongoing meetings as well as upcoming meetings section

mkesavan13 commented on app.js:
```diff
@@ -0,0 +1,131 @@
+    
+    const webhookApiUrl = 'https://webexapis.com/v1/meetings';
+    
+    async function fetchMeetings() {
```
It is better to have all these function definition out of DOMContentLoaded event and just use this event to call `fetchMeetings` method.

mkesavan13 reviewed (changes requested):
Please address the review comments and then we are good to merge

Parimala032 committed:
fix: Added structured HTML to display meeting details line by line

set empty action for join button

created separate function for list meetings

Moved all function definitions out of the DOMContentLoaded event. Updated DOMContentLoaded event to only call the fetchMeetings method

Parimala032 committed:
fix: Removed the default SIP address from the meeting details

mkesavan13 commented on app.js:
```diff
@@ -0,0 +1,146 @@
+        listElement.appendChild(meetingElement);
+    });
+}
+function createMeetingList(meetings, filterFn, listElement, emptyMessage) {
```
I can see this method written twice?

mkesavan13 reviewed (approved):
Looks good to me. Just one comment. Once that is addressed, I can merge it.

Parimala032 committed:
fix:Removed the duplicate createMeetingList function
